compiler/rustc_lint/src/levels.rs

```diff
@@ -1028,6 +1028,8 @@ impl<'s, P: LintLevelsProvider> LintLevelsBuilder<'s, P> {
         }
 
         if !is_crate_node {
+            // We break on the first lint, and it does not matter which one.
```

Comment:
It may not matter, but emitting a different diagnostic depending on the architecture is a bug.

Reply:
I changed `current_specs()` to return a `FxIndexMap`, which only cascaded to 13 other places, which seems fine. So now the iteration order here should be deterministic.

(Due to conflicts I also had to rebase.)